dcrpool.go

```diff
@@ -123,7 +123,51 @@ func newPool(cfg *config) (*miningPool, error) {
 	if err != nil {
 		return nil, err
 	}
-	err = p.hub.Connect()
+
+	// Establish a connection to the mining node.
+	ntfnHandlers := p.hub.CreateNotificationHandlers()
+	nodeConn, err := rpcclient.New(dcrdRPCCfg, ntfnHandlers)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := nodeConn.NotifyWork(); err != nil {
+		nodeConn.Shutdown()
+		return nil, err
+	}
+	if err := nodeConn.NotifyBlocks(); err != nil {
+		nodeConn.Shutdown()
+		return nil, err
+	}
+
+	p.hub.SetNodeConnection(nodeConn)
+
+	// Establish a connection to the wallet if the pool is mining as a
+	// publicly available mining poolin solo.
+	if !cfg.SoloPool {
+		creds, err := credentials.
+			NewClientTLSFromFile(cfg.WalletRPCCert, "localhost")
+		if err != nil {
+			return nil, err
+		}
+
+		grpc, err := grpc.Dial(cfg.WalletGRPCHost,
```

Comment:
As far as I can tell, this grpc client can be [safely accessed concurrently](https://github.com/grpc/grpc-go/blob/master/Documentation/concurrency.md) and does not need to be protected by a mutex.

I believe the same is also true for the dcrd client, but I'm not 100% sure on that.